test/unit/mdc-chips/mdc-chip.foundation.test.js

```diff
@@ -99,6 +104,70 @@ test('on click, emit custom event', () => {
   td.verify(mockAdapter.notifyInteraction());
 });
 
+test('on leading icon opacity transition end, add `mdc-chip__icon--hidden-leading`' +
+  'class to leading icon if chip is selected', () => {
+  const {foundation, mockAdapter} = setupTest();
+  td.when(mockAdapter.hasClass(cssClasses.SELECTED)).thenReturn(true);
+  const handlers = captureHandlers(mockAdapter, 'registerLeadingIconEventHandler');
+  const mockEvt = {
+    type: 'transitionend',
+    propertyName: 'opacity',
+  };
+
+  foundation.init();
+  handlers.transitionend(mockEvt);
+
+  td.verify(mockAdapter.addClassToLeadingIcon(cssClasses.HIDDEN_LEADING_ICON));
+});
+
+test('on leading icon opacity transition end, add `mdc-chip__icon--hidden-leading`' +
+  'class to leading icon if chip is not selected', () => {
+  const {foundation, mockAdapter} = setupTest();
+  td.when(mockAdapter.hasClass(cssClasses.SELECTED)).thenReturn(false);
+  const handlers = captureHandlers(mockAdapter, 'registerLeadingIconEventHandler');
+  const mockEvt = {
+    type: 'transitionend',
+    propertyName: 'opacity',
+  };
+
+  foundation.init();
+  handlers.transitionend(mockEvt);
+
+  td.verify(mockAdapter.addClassToLeadingIcon(cssClasses.HIDDEN_LEADING_ICON), {times: 0});
+});
+
+test('on checkmark opacity transition end, remove `mdc-chip__icon--hidden-leading`' +
+  'class from leading icon if chip is not selected', () => {
+  const {foundation, mockAdapter} = setupTest();
+  td.when(mockAdapter.hasClass(cssClasses.SELECTED)).thenReturn(false);
+  const handlers = captureHandlers(mockAdapter, 'registerCheckmarkEventHandler');
+  const mockEvt = {
+    type: 'transitionend',
+    propertyName: 'opacity',
+  };
+
+  foundation.init();
+  handlers.transitionend(mockEvt);
+
+  td.verify(mockAdapter.removeClassFromLeadingIcon(cssClasses.HIDDEN_LEADING_ICON));
+});
+
+test('on checkmark opacity transition end, remove `mdc-chip__icon--hidden-leading`' +
```

Comment:
This test name seems inaccurate if we're verifying that the class _isn't_ removed?